### DISCUSSION TITLE
Has anyone used ZuluSCSI on a SPARCStation?

### DISCUSSION THREAD

**Discussion by mpryon** [ZuluSCSI/ZuluSCSI-firmware#122] [Q&A] [answered]
Hiya,

I've got my ZuluSCSI installed in a sparcstation 2 and while it's definitely working...something is amiss.  I can boot from CD image and Solaris 2.7 sees the hard drive image but I've having some issues I cannot sort.

I can setup partitions on the image but thus far I've not been able to write anything else to the card.

I've fiddled with termination settings, used a physical CD drive instead of an image, tried several different SD cards and while all of my blind stumbling changes bevahiour somewhat, I'm still not able to install Solaris 2.7 or SunOS 4.1.4.

I was wondering if anyone has done so successfully and could offer any pointers?

thanks!

**Comment by PetteriAimonen** [comment, marked as answer]
I have had some emails about a SparcStation 20. There the installation failed with "Could not update disks with new configuration". This was my response to that case:

> The debug log appears completely normal and all commands appear to be succeed.
> 
> Googling finds this thread with similar problems:
> https://community.oracle.com/tech/apps-infra/discussion/1932859/solaris-7-cannot-install-file-system-creation-failed
> 
> The idea that Solaris is having trouble creating their proprietary VTOC partition table sounds plausible.
> Using fmthard (https://docs.oracle.com/cd/E88353_01/html/E72487/fmthard-8.html) to manually reformat the drive might be
> worth a try, if that is possible to do from the install CD.

Are you receiving any particular error messages?
Can you turn the debug switch on and then post resulting `zululog.txt` after the problem has occurred?

**Reply by preryon** [reply]
It looked like it failed early on making the file systems but it went by quickly.
Solaris 7 install does continue strangely but it's an endless stream of "Sense Key: Media Error".

Later today I'll try again and flip the debug switch. I'll get back to you that that log.

thanks!

![PXL_20221216_182646142](https://user-images.githubusercontent.com/45078007/208168549-685cf26b-f713-4f43-870a-97f8aaf862a9.jpg)


n.B. I was getting some errors initially when booting from a CD image. I renamed the Solaris ISO CD60_512.hda and the CD seems to be working without error now.

**Reply by preryon** [reply]
Well......that's funny....I got distracted by a couple of fires at work and forgot the Sparc was installing. Several hours later and page after page after page of the above write error, the install finished and booted!

**Reply by preryon** [reply]
I'll still happily do a debug capture if you think it'll help?

**Reply by preryon** [reply]
I may have spoken too soon. There were plenty of errors and the system isn't exactly usable but it obviously wrote to the disk well enough to boot up and present the GUI login.

**Reply by preryon** [reply]
The plot thickens...the SD card that sort of succeeded with an install? I'd forgotten the I had previously installed a sun4m build of Solaris 7 in qemu on the image file. The SS2 cannot boot from that as it's the wrong architecture but the installer then sort of successfully installed over the already created file system.

I've just tried again to install on another image file that hadn't been previously formatted and sure enough, it failed to create the file system.

Preparing system for Solaris install

Configuring disk (c0t3d0)
        - Creating Solaris disk label (VTOC)

Creating and checking UFS file systems
        - Creating / (c0t3d0s0)

ERROR: File system creation failed for / (c0t3d0s0)

ERROR: Could not check or create system critical file systems

ERROR: Could not update disks with new configuration

Could not update disks with new configuration


I did turn on debug and will grab the log in a little bit.

**Reply by preryon** [reply]
[zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/10250927/zululog.txt)

**Reply by PetteriAimonen** [reply]
Is that log from the formatting? No errors visible there.

Some systems are quite picky about image sizes and sector counts when formatting drives, so it might help if we knew some model of harddrive that was originally supported by the system.

**Reply by PetteriAimonen** [reply]
Oh, and regarding that earlier media error: that would typically mean an error when writing to the SD card. If that happens, there should be more information in the `zululog.txt`, regardless whether debug was enabled.

**Reply by preryon** [reply]
Yep, that log was from an attempt to setup the drive and install.

I'd not be surprised if the sparc is picky about geometry. I'm trying to find some specs on drives of the era.

One thing for sure, the SS2 expects parity to be enabled on SCSI drives. Is it possible to turn parity checking on/off on ZuluSCSI?

**Reply by jamadden** [reply]
> I'd not be surprised if the sparc is picky about geometry. I'm trying to find some specs on drives of the era.

According to the [SS2 field service manual](https://docs.oracle.com/cd/E19127-01/sparc2.ws/800-5166-10/800-5166-10.pdf) the original part was a 213MB drive with Sun part number 370-1327; that appears to be a [Maxtor model LXT213SY](https://wiki.preterhuman.net/Maxtor_LXT213SY):
```
Capacity Unformatted:
        per drive        248   MB
        per surface         35.4 MB

Capacity formatted:
        per drive        212,703,232
        per surface         30,386,176

                                Cylinders per zone        Sectors per Track
formatted disk data rates:        Physical        Logical        Physical  Logical
Zone 1        1.69MB/s                220                214        56           55
     2        1.60MB/s                220                219        53           52
     3  1.47MB/s                220                219        49           48
     4  1.32MB/s                220                219        45           44
     5  1.23MB/s                220                219        41           40
     6  1.01MB/s                220                219        34           33

Tracks per zone are 1505 in zone 1 and 1533 in all others.

Rotational speed is        3600rpm
Recording method is        1,7RLL
Track Density is        1 600 tpi
Data heads:                7
Servo heads:                1
Disks                        4
```

For something bigger, I also have three different size/age drives that work in a Sparcstation 20; I know it's not exactly the same architecture as a SS2, and the drives are SCA instead of 50-pin, but in case it helps, the oldest/smallest/OEM 4GB drive is an IBM model DCAS-34330. According to [here](https://www.manualslib.com/manual/574162/Ibm-Dcas-32160-Ultrastar-2-1-Gb-Hard-Drive.html?page=23#manual), that's 512 byte sectors, with 132-211 sectors per track, 6 heads, 3 disks.

(I haven't tried a Zulu in the 20 yet. I'd like to, but the SCA vs 50-pin is blocking me right now.)

**Reply by PetteriAimonen** [reply]
@preryon All ZuluSCSI boards send parity bits to host, so that should be fine. ZuluSCSI v1.1 does not currently check parity of data received from host.

The geometry can be set in `zuluscsi.ini`. The image file size should then be divisible by the number of logical sectors and heads. Though in the log there is no visible request for geometry information, so it might be that the SPARCStation does not care.

**Reply by preryon** [reply]
An update and spoiler, it's a qualified success.

I obtained a 4.5gb SCSI drive and installed Solaris 2.7 from a CD image on the ZuluSCSI. No issues. Win.

I made a 4.5gb disk image and tried in install from the the CD image and it failed in the same was as previous tries.

I renamed the 4.5 image to HD1.hda and booted from the real SCSI disk.
I dd'd the hard drive to the image and then renamed the image HD3.hda.

It boots up and works perfectly (as far as I can tell in limited testing).

Obviously, it'd be nice to install directly to an image but it looks promising.

**Reply by aperezbios** [reply]
@preryon great, thanks for the update! Glad it's working out well.

**Reply by PetteriAimonen** [reply]
I think the issue may be that full / low-level reformatting or repartitioning is not working for some reason. Possibly due to geometry settings or something else, which causes Sparc to get confused about what kind of disk it is.

**Reply by mpryon** [reply]
That does seem likely...a previous experiment that _nearly_ worked was formatting the disk image in qemu and then using it on the real sparc in ZuluSCSI. The Solaris install did work but there were loads of write errors. When the install finished and booted, I couldn't log in. It seemed the shadow file was trashed.
